modules/yieldmoBidAdapter.js

```diff
@@ -0,0 +1,153 @@
+      placement.placement_id = bid.placementCode;
+      placement.sizes = bid.sizes;
+
+      placements.push(placement);
+    }
+
+    url = utils.tryAppendQueryString(url, 'p', JSON.stringify(placements));
+    return url;
+  }
+
+  function _appendImpressionInformation(url) {
+    var page_url = document.location; // page url
+    var pr = document.referrer || ''; // page's referrer
+    var dnt = (navigator.doNotTrack || false).toString(); // true if user enabled dnt (false by default)
+    var _s = document.location.protocol === 'https:' ? 1 : 0; // 1 if page is secure
+    var description = _getPageDescription();
+    var title = document.title || ''; // Value of the title from the publisher's page. 
+    var bust = new Date().getTime().toString(); // cache buster
+    var scrd = window.devicePixelRatio || 0; // screen pixel density
+
+    url = utils.tryAppendQueryString(url, 'callback', '$$PREBID_GLOBAL$$.YMCB');
+    url = utils.tryAppendQueryString(url, 'page_url', page_url);
+    url = utils.tryAppendQueryString(url, 'pr', pr);
+    url = utils.tryAppendQueryString(url, 'bust', bust);
+    url = utils.tryAppendQueryString(url, '_s', _s);
+    url = utils.tryAppendQueryString(url, 'scrd', scrd);
+    url = utils.tryAppendQueryString(url, 'dnt', dnt);
+    url = utils.tryAppendQueryString(url, 'description', description);
+    url = utils.tryAppendQueryString(url, 'title', title);
+
+    return url;
+  }
+
+  function _getPageDescription() {
+    if (document.querySelector('meta[name="description"]')) {
+      return document.querySelector('meta[name="description"]').getAttribute('content'); // Value of the description metadata from the publisher's page.  
+    } else {
+      return '';
+    }
+  }
+
+  // expose the callback to the global object:
+  $$PREBID_GLOBAL$$.YMCB = function(ymResponses) {
+    if (ymResponses && ymResponses.constructor === Array && ymResponses.length > 0) {
+      for (var i = 0; i < ymResponses.length; i++) {
+        _registerPlacementBid(ymResponses[i]);
+      }
+    } else {
+      // If an incorrect response is returned, register error bids for all placements
+      // to prevent Prebid waiting till timeout for response 
+      _registerNoResponseBids();
+
+      // @if NODE_ENV='debug'
+      utils.logMessage('No prebid response for placement %%PLACEMENT%%');
+      // @endif
+    }
+  };
+
+  function _registerPlacementBid(response) {
+    var bidObj = utils.getBidRequest(response.callback_id);
+    var placementCode = bidObj && bidObj.placementCode;
+    var bid = [];
+
+    if (response && response.cpm && response.cpm !== 0) {
+      bid = bidfactory.createBid(1, bidObj);
+      bid.bidderCode = 'yieldmo';
+      bid.cpm = response.cpm;
+      bid.ad = response.ad;
+      bid.width = response.width;
+      bid.height = response.height;
+      bidmanager.addBidResponse(placementCode, bid);
+    } else {
+      // no response data
+      // @if NODE_ENV='debug'
```

Comment:
Prebid isn't using `preprocess`... so it won't respect these annotations.

The `utils.logMessage` function is a no-op unless debugging is enabled, though.

Reply:
👍